test/extensions/filters/listener/common/fuzz/listener_filter_fakes.h

```diff
@@ -0,0 +1,95 @@
+#include "common/api/os_sys_calls_impl.h"
+#include "common/network/io_socket_handle_impl.h"
+#include "common/network/utility.h"
+
+#include "test/mocks/network/mocks.h"
+
+#include "gmock/gmock.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+
+class FakeConnectionSocket : public Network::MockConnectionSocket {
+public:
+  FakeConnectionSocket()
+      : io_handle_(std::make_unique<Network::IoSocketHandleImpl>(42)), local_address_(nullptr),
+        remote_address_(nullptr) {}
+
+  ~FakeConnectionSocket() override { io_handle_->close(); }
+
+  Network::IoHandle& ioHandle() override { return *io_handle_; }
+
+  const Network::IoHandle& ioHandle() const override { return *io_handle_; }
+
+  void setLocalAddress(const Network::Address::InstanceConstSharedPtr& local_address) override {
+    local_address_ = local_address;
+    if (local_address_ != nullptr) {
+      addr_type_ = local_address_->type();
+    }
```

Comment:
Should we be clearing `addr_type_` if the provided local_address is null?

Reply:
I'm not sure how to clear it, since I think addr_type_ has to be set. Would it make more sense to add checks to make sure that local_address_ isn't a nullptr elsewhere (such as in ipVersion)?

Reply:
Yeah, having an ASSERT anywhere we access information about local_address_ sounds like a good idea.